connector/exceptionsconnector/connector_metrics.go

```diff
@@ -132,28 +134,42 @@ func (c *metricsConnector) collectExceptions(ilm pmetric.ScopeMetrics) error {
 	dps := mCalls.Sum().DataPoints()
 	dps.EnsureCapacity(len(c.exceptions))
 	timestamp := pcommon.NewTimestampFromTime(time.Now())
-	for _, val := range c.exceptions {
-		dpCalls := dps.AppendEmpty()
-		dpCalls.SetStartTimestamp(c.startTimestamp)
-		dpCalls.SetTimestamp(timestamp)
-
-		dpCalls.SetIntValue(int64(val.count))
-
-		val.attrs.CopyTo(dpCalls.Attributes())
+	for _, exc := range c.exceptions {
+		dp := dps.AppendEmpty()
+		dp.SetStartTimestamp(c.startTimestamp)
+		dp.SetTimestamp(timestamp)
+		dp.SetIntValue(int64(exc.count))
+		for i := 0; i < exc.exemplars.Len(); i++ {
+			exc.exemplars.At(i).SetTimestamp(timestamp)
+		}
+		exc.exemplars.CopyTo(dp.Exemplars())
+		exc.attrs.CopyTo(dp.Attributes())
 	}
 	return nil
 }
 
-func (c *metricsConnector) addException(excKey string, attrs pcommon.Map) {
+func (c *metricsConnector) addException(excKey string, attrs pcommon.Map) *exception {
 	exc, ok := c.exceptions[excKey]
 	if !ok {
-		c.exceptions[excKey] = &excVal{
-			count: 1,
-			attrs: attrs,
+		c.exceptions[excKey] = &exception{
+			count:     1,
+			attrs:     attrs,
+			exemplars: pmetric.NewExemplarSlice(),
 		}
-		return
+		return c.exceptions[excKey]
 	}
 	exc.count++
+	return exc
+}
+
+func (c *metricsConnector) addExemplar(exc *exception, traceID pcommon.TraceID, spanID pcommon.SpanID) {
+	if !c.config.Exemplars.Enabled || traceID.IsEmpty() {
+		return
+	}
+	e := exc.exemplars.AppendEmpty()
```

Comment:
Can this Append result in an unbounded growth and that in fact might be the root cause for the panic I noticed in my overnight test. 
So, an alternate approach could be to reset exc.exemplars to a new slice after `exc.exemplars.CopyTo(dp.Exemplars())`

Reply:
I think we could just add that count mechanism that spansmetricsconnector is using, wdyt?

Reply:
Hello Marc - That should be good IMO. However, what do you think of resetting `exc.exemplars` back to `pmetric.NewExemplarSlice()` after doing `CopyTo`
I guess I do not understand the need to hang on to older exemplars post that CopyTo operation. Can you please help me understand?

Reply:
You are totally right, what about now?

Reply:
Looks good to me Marc. Thank you.